contract-sdk/src/lib.rs

```diff
@@ -52,6 +49,26 @@ pub struct BlobIndex(pub u32);
 #[derive(Default, Serialize, Deserialize, Debug, Clone, PartialEq, Eq, Hash, Encode, Decode)]
 pub struct BlobData(pub Vec<u8>);
 
+#[derive(Debug, Serialize, Deserialize, Default, Clone, PartialEq, Eq, Encode, Decode, Hash)]
+pub struct Blob {
+    pub contract_name: ContractName,
+    pub data: BlobData,
+}
+
+pub fn flatten_blobs(blobs: &[Blob]) -> Vec<u8> {
+    blobs
+        .iter()
+        .flat_map(|b| {
+            let mut combined = b.contract_name.0.clone().into_bytes();
+            combined.extend(b.data.0.clone());
```

Comment:
mem copy is costly in guest, can we have an impl of this without cloning ? (not necessary to spend too much time on it, it's not that relevant for now)